runtime/amd64.S

```diff
@@ -1024,8 +1024,7 @@ ENDFUNCTION(G(caml_runstack))
 
 FUNCTION(G(caml_ml_array_bound_error))
 CFI_STARTPROC
-        ENTER_FUNCTION
-        LEA_VAR(caml_array_bound_error, %rax)
```

Comment:
If anyone wonders: `ENTER_FUNCTION` maintains `%rbp` as frame pointer in "with frame pointers" mode, and does nothing otherwise.  I guess it would have to be matched by a `LEAVE_FUNCTION` just before tail-calling `caml_c_call`.  @stedolan chose to omit the `ENTER_FUNCTION` / `LEAVE_FUNCTION` dance entirely, which is correct too and probably just as good in practice.  

(If I understand correctly, the only case where it might make a difference is in frame-pointer mode, if a profiling interrupt hits while executing the `LEA_VAR` instruction, in which case the `%rbp`-based backtrace will not be accurate.  Nothing to worry about.)

Reply:
Actually, `ENTER_FUNCTION` seems required to get a correct backtrace in frame-pointer mode.